sanic/http.py

```diff
@@ -492,7 +492,7 @@ async def read(self) -> Optional[bytes]:
                 self.request_body = None
                 # Because we are leaving one CRLF in the buffer, we manually
                 # reset the buffer here
-                self.recv_buffer = bytearray()
+                self.recv_buffer = self.protocol.recv_buffer = bytearray()
```

Comment:
This is still incorrect, and in general you should not recreate the buffer. Use `del self.recv_buffer[:2]` or so to keep the existing buffer but to consume the data that you want removed (note: as discussed earlier, removing all or the first two bytes here might be invalid anyway).

Reply:
I still am not sure I see how. If we do not, and even if we account for the http loop, you end up with `\r\n` as the first two bytes on the next request.